src/puppetlabs/trapperkeeper/services/status/status_service.clj

```diff
@@ -0,0 +1,30 @@
+(ns puppetlabs.trapperkeeper.services.status.status-service
+  (:require [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [puppetlabs.trapperkeeper.services :refer [service-context]]
+            [puppetlabs.trapperkeeper.services.status.status-core :as core]
+            [puppetlabs.trapperkeeper.services.webrouting.webrouting-service :as webrouting-service]))
+
+(defprotocol StatusService
+  (register-status [this service-name service-version status-version status-fn]
+    "Register a status callback function for a service by adding it to the
+    status service context."))
+
+(defservice status-service
+  StatusService
+  [[:WebroutingService add-ring-handler]]
+
+  (init [this context]
+    (assoc context :status-fns (atom {})))
+
+  (start [this context]
+    (log/info "Registering status service HTTP API at /status")
+    (let [handler (core/build-handler context)]
```

Comment:
I'd prefer the status functions to be extracted from `context` here and then passed-in to `build-handler`, as opposed to passing the TK context around.  It looks a bit confusing (especially since compojure has its own context) and the core functions just need those functions, so seeing that in arg lists makes it a bit more obvious what's going on.  Yes, this is a nitpick.  :nail_care: